src/msg/Dispatcher.h

```diff
@@ -204,13 +204,16 @@ class Dispatcher {
   /**
    * handle successful authentication (msgr2)
    *
-   * Authenticated result/state will be attached to the Connection.
+   * Authenticated result/state will be attached to the Connection. This is
+   * called via the MonClient.
+   *
+   * Do not acquire locks in this method! It is considered "fast" delivery.
    *
    * return 1 for success
    * return 0 for no action (let another Dispatcher handle it)
    * return <0 for failure (failure to parse caps, for instance)
```

Comment:
> I've already gone through other users of `ms_handle_authentication` [...]

Have you looked into why most callers seem to ignore the failure?

Reply:
That's a great question. I did not. It appears the only side-effect is that caps which fail to parse can continue talking to the daemon but e.g. for the mons it just treats it as unauthenticated with no privileges.

I will create a new ticket to clean this up. I don't think it necessarily needs to be in this PR.

Reply:
https://tracker.ceph.com/issues/62382

Reply:
I will work on that tomorrow.